Prevents Select tool from erasing redo stack

bsb20 opened the pull request:
```
-Tracks Select tool state to wait until an actual change is made
-Only allows a Select operation to trigger a save when a change is
made
```

Commit:
Prevents Select tool from erasing redo stack

    -Tracks Select tool state to wait until an actual change is made
    -Only allows a Select operation to trigger a save when a change is
    made

cwillisf commented on src/svgeditor/BitmapEdit.as:
```diff
@@ -136,10 +136,11 @@ public class BitmapEdit extends ImageEdit {
 	}
 
 	protected override function selectHandler(evt:Event = null):void {
```
For the sake of readability, could I get you to put a line at the top like `var foo:ObjectTransformer = currentTool as ObjectTransformer;` (with a better name than `foo` of course) and then simplify the rest of the function? I know you're not the one that started the fire but it seems like it's continuing to spread ;)

cwillisf commented:
:+1:, though if you wouldn't mind cleaning up the `is`/`as` stuff it'd be even better :)

Commit:
Refactored is/as ObjectTrasnformer checks

    -Created a read-only property which performs this cast on
    currentTool
    -Only replaced places where is/as checks were done explicitly
    -This means checks like `toolMode == "select"` were left in place.
    This is precautionary, and would probably be safe to changes
    eventually.

bsb20 commented:
I changed this to use a property the way `segmentationTool` is used, and I only did the most basic replacements.

cwillisf commented:
Oh! Sorry, I see now how my comments might have been misleading. I only meant to suggest `is`/`as` changes within the `selectHandler()` function in `BitmapEdit.as`. I didn't mean to make you do as much as you did!
I do think your changes made `ImageEdit.as` nicer to read, though!
